Monitor query updates (#17459)
Azure/azure-sdk-for-js: sdk/monitor/monitor-query/README.md

|---statistics

> |---visalization

Reviewer: ```suggestion
|---visualization
```

|---error

> |--- logsQueryResultStatus ("Partial" | "Success" | "Failed")

Reviewer: ```suggestion
|---logsQueryResultStatus ("Partial" | "Success" | "Failed")
```

  const tokenCredential = new DefaultAzureCredential();
  const logsQueryClient = new LogsQueryClient(tokenCredential);

>   const kqlQuery = "AppEvents | project TimeGenerated, Name, AppRoleInstance | limit 1";

Author: Kusto Query language. I have the link here under the concepts - https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/monitor/monitor-query Should that be sufficient?

```
LogsQueryBatchResult
|---results (list of following objects)
    |---status
    |---statistics

>     |---visalization

Reviewer: ```suggestion
    |---visualization
```

    |---error
    |---logsQueryResultStatus ("Partial" | "Success" | "Failed")
    |---tables (list of `LogsTable` objects)
        |---name
        |---rows

>         |---columns (list of `LogsColumn` objects)

Reviewer: Should `columns` be renamed to `columnDescriptors`?

      }
    }
  }
  // next query
  i++;
}
```

A full sample can be found [here](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/monitor/monitor-query/samples/v1/typescript/src/logsQueryBatch.ts)

> **Throttling Guidance Log Queries**
> 
> Please refer the [service guidance for throttling](https://dev.loganalytics.io/documentation/Using-the-API/Limits) for API calls.

Reviewer: ```suggestion
For information on request throttling at the Log Analytics service level, see [Rate limits](https://dev.loganalytics.io/documentation/Using-the-API/Limits).
```

});
```

> In the above sample, the ordering of results for the metrics in the `metricResponse` will be in the order in which the user specifies the metric names in the`metricNames` array argument for the query method. If user specifies - `[firstMetric.name,secondMetricName]`
> The result for `firstMetric.name` will appear before the result for `secondMetricName` in the `metricResponse`

Reviewer: ```suggestion
In the preceding sample, the ordering of results for the metrics in the `metricResponse` will be in the order in which the user specifies the metric names in the `metricNames` array argument for the query method. If user specifies `[firstMetric.name,secondMetricName]`, the result for `firstMetric.name` will appear before the result for `secondMetricName` in the `metricResponse`.
```

    |---timeseries (list of `TimeSeriesElement` objects)
        |---metadataValues
        |---data (list of data points represented by `MetricValue` objects)

> |---getMetricByName(metricName): Metric (convenience method)

Reviewer: We should also have a sample somewhere for this new function.

Author: I have added these two lines to the existing sample - 
```
const metric = metricsResponse.getMetricByName(firstMetric.name!);
console.log(`Selected Metric: ${firstMetric.name}`, JSON.stringify(metric, undefined, 2));
```

);
```

> In order to view the results for each workspace, you can use the `TenantId` column to either order the results or filter them in the Kusto query.

Reviewer: ```suggestion
To view the results for each workspace, use the `TenantId` column to either order the results or filter them in the Kusto query.
```

> Order results by TenantId -

Reviewer: ```suggestion
**Order results by TenantId**
```

```
AppEvents | order by TenantId
```

> Filter results by TenantId

Reviewer: ```suggestion
**Filter results by TenantId**
```
